views/forgot.go

```diff
@@ -29,67 +29,65 @@ var notification = Notification{
 
 // ForgotFunc handles sending a reset email
 func (v *Views) ForgotFunc(c echo.Context) error {
-	var err error
 	switch c.Request().Method {
 	case "GET":
 		return v.template.RenderTemplate(c.Response().Writer, nil, templates.ForgotTemplate, templates.NoNavType)
 	case "POST":
-		err = c.Request().ParseForm()
-		if err != nil {
-			http.Error(c.Response(), err.Error(), http.StatusBadRequest)
-		}
-		u := user.User{Email: c.Request().Form.Get("email")}
+		u := user.User{Email: c.FormValue("email")}
 
 		if u.Email == "" {
 			return v.template.RenderTemplate(c.Response(), nil, templates.ForgotTemplate, templates.NoNavType)
 		}
 		// Get user and check if it exists
-		user1, err := v.user.GetUser(c.Request().Context(), u)
+		userFromDB, err := v.user.GetUser(c.Request().Context(), u)
 		if err != nil {
 			// User doesn't exist, we'll pretend they've got an email
-			log.Printf("request for reset on unknown email \"%s\"", user1.Email)
+			log.Printf("request for reset on unknown email \"%s\"", userFromDB.Email)
 			return v.template.RenderTemplate(c.Response(), notification, templates.NotificationTemplate, templates.NoNavType)
 		}
 		url := uuid.NewString()
-		v.cache.Set(url, user1.UserID, cache.DefaultExpiration)
+		v.cache.Set(url, userFromDB.UserID, cache.DefaultExpiration)
+
+		mailer := mail.NewMailer(mail.Config{
```

Comment:
Bordering on a NITPICK - rather than creating a new mailer and passing around the config each time, it might make sense to have a persistent `*Mailer` in `Views`, which hangs onto its config (seeing as it's not going to change at runtime) and takes care of connecting and disconnecting.

Reply:
This has been changed now